firebase-android-release-dashboard/src/components/Release/ReleaseCard/ReleaseCard.js

```diff
@@ -9,13 +10,21 @@ import useStyles from "./styles.js";
  * The ReleaseCard component renders a card containing the metadata
  * and details of a release.
  *
- * @param {Object} release - The details of the release to render.
- * @throws {Error} If the release state is not recognized.
+ * The release is fetched from Firestore using the useRelease hook.
+ * We don't pass the whole release as a prop to the ReleaseCard
+ * component because we want to be able to have the component
+ * re-render when the release changes.
  *
+ * @param {Object} releaseId - The id of the release to render.
  * @return {JSX.Element} The rendered component.
  */
-function ReleaseCard({release}) {
+function ReleaseCard({releaseId}) {
   const classes = useStyles();
+  const release = useRelease(releaseId);
+
+  if (!release) {
+    return null;
```

Comment:
Is this expected to happen? Is there a placeholder that should be here? Or perhaps this should throw an error if we never expect it to occur?

Reply:
This is expected to happen for a very brief period while the asynchronous fetch from Firestore hasn't been resolved yet. You're right though, this should have a placeholder, so I added a circular progress icon to be displayed while this fetch is happening
https://github.com/firebase/firebase-release-dashboard/pull/21/commits/3c01c6cc995cfc8357f8ec0d3cb27fd7e59967f3